dooboolab/dooboo-ui-native
# [Feature][Branch] Select -> "Root" section

## Description

### 🙋🏻‍♂️ please do not merge this until making TODO done

![root section in feat/select](https://user-images.githubusercontent.com/33364619/70374774-fd328c00-1939-11ea-9af4-776c6885bd41.png)
![스크린샷 2019-12-10 오후 4 54 47](https://user-images.githubusercontent.com/19166187/70506573-0015c280-1b6e-11ea-8f8a-b256852a6e3f.png)

- add **theme** prop: `none`, `box`, `underline`
- add **rootViewStyle** prop: custom `ViewStyle` stylesheet to root part of selectbox's view
- add **rootTextStyle** prop: custom `TextStyle` stylesheet to root part of selectbox's text

## Related Issues

#60 

## Tests

- check theme and rootTextStyle and placeholder with case "notheme"
- check theme and rootTextStyle and placeholder with case "inputTheme"
- check theme and rootViewStyle and placeholder with case "themeAndRootView"
- check theme & rootViewStyle & rootTextTheme and placeholder with case "disabled"

## TODO

- [x] make styling **box-shadow** looking good
- [x] typechecking using `tsc`
- [x] making test code related this

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/dooboo-ui-native/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
